### ISSUE TITLE
Issue connecting Dapps to my forked & customized version of metamask.

### ISSUE THREAD

**Issue by theiceeman** [MetaMask/providers#207]
Hi,
I forked the Metamask wallet repo, and have been adding some features I believe will make the application even more powerful.
I'm trying to connect to my updated version of the wallet using live Dapps but they can't seem to distinguish between the two wallets, when installed on the same browser.
I believe its a provider issue, and i've been working to find a way to distinguish the wallets as separate wallets to Dapps to no avail.
[This](https://github.com/MetaMask/providers/blob/main/src/BaseProvider.ts) is the `metamask/provider` package which handles the connection by Dapps to Metamask in this file `Baseprovider.ts`.

I found the `handleConnect` function inside, but can't seem to make sense of how it handles the connection by Dapps, and how I can edit it to suit my needs.
I'm hoping someone that better understands the project can point me to what I need to change to achieve this.
I’ll really appreciate any help.
Thanks.

**Comment by mcmire** [comment]
Is there a reason why you want your dapp (or any dapp) to support multiple versions of MetaMask which are installed at the same time? We don't support this workflow and would recommend that you keep one version of the extension enabled. Is that possible?

**Comment by theiceeman** [comment]
Hi @mcmire , 
Thanks for replying. 
So i have two wallet extensions: 
one is metamask, and another one is my forked version of metamask. 
I want to let users use both wallets on the same browser. 
My Dapp officially uses the forked Metamask wallet, but still users should be able to connect with metamask if they choose. 
So from the users side, when they want to connect with a web3 wallet, i need to identify which one is metamask and which one is my forked version.
I believe this is possible. 
Thanks.

**Comment by mcmire** [comment]
Yeah, unfortunately, it just isn't possible to have two different versions of MetaMask installed right now, even if one is a forked version, and we don't have plans to support this any time soon. If there's some improvements you'd like to make to MetaMask, however, feel free to submit them to our [extension repo](https://github.com/MetaMask/metamask-extension). Or, if you're testing our your forked version, you can simply go to your Extensions page (`chrome://extensions` in Chrome, `about://addons` in Firefox) and disable the official version of MetaMask instead of having to outright uninstall it. Hope that helps.

**Comment by aqeelVaival** [comment]
Hi folks,
Is there any further development on this? or still we have the same situation?

**Comment by Chiefy0x1** [comment]
https://github.com/MetaMask/metamask-mobile/issues/4895
